OneCommunityGlobal/HighestGoodNetworkApp
# Jerry - Combined changes 1199+1281

# Description
The main motivation of this PR is to combine FE PRs 1199 and 1281, which happen to make changes to many of the same files and lines of code. To avoid a merge conflict when PRs 1199 and 1281 get merged into development branch, this PR was created.

## Related PRS (if any):
Backend: BE PR #[539](https://github.com/OneCommunityGlobal/HGNRest/pull/539)
Frontend: #1199 and #1281 

## Main changes explained:
- Employing consistent naming of permissions in `src/utils/routePermissions.js`
- Deciding that when a volunteer account has both `See User Management Tab (Full Functionality)` AND `See User Management Tab (ONLY create Users)`, then the `See User Management Tab (Full Functionality)` permission will have priority. I.e., the volunteer account will be able delete, edit, and create users. 


## How to test:
1. In backend (BE PR #[539](https://github.com/OneCommunityGlobal/HGNRest/pull/539)) check into `jerry_user_management_permissions` branch and run the branch locally.
2. In frontend, check into current branch.
3. do `npm install` and `...` to run this PR locally
4. Log in as Owner -> Other LInks -> Permissions Management -> Manage User Permissions
5. Enter the username of an account that does not have the “See User Management Tab (Full Functionality)” permission by default (i.e., a Volunteer account). 
6. Next to “See User Management Tab (Full Functionality)” or “See User Management Tab (ONLY create Users)”, click on Add. Then scroll down and click on Submit.
7. Log in as the volunteer account (if already logged in, make sure to log out and log back in). The “Reports” button should appear with ONLY the “User Management” option visible/accessible.
8. The volunteer account can create, delete, and update users in the User Management page if "See User Management Tab (Full Functionality" permission was added. If only the "See User Management Tab (ONLY create Users)" permission was added, then the volunteer can only create users.

## Screenshots or videos of changes:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/55968519/f3d46882-81c5-4932-b8e3-079823196667


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/55968519/350578ce-a743-48df-9ef4-8e3e427e3f6e
